File: java/freeman/command/undo/CeilingFan.java
public class CeilingFan {
    public static final int HIGH=3;
    public static final int MEDIUM=2;
    public static final int LOW=1;
    public static final int OFF=0;
    private int speed;
    private String location;
    public CeilingFan(String location){
        this.location=location;
        speed=OFF;
    }
    public void high(){
        speed=HIGH;
    }
    public void medium(){
        speed=MEDIUM;
    }
    public void low(){
        speed=LOW;
    }
    public void off(){
        speed=OFF;
    }
    public int getSpeed(){
        return speed;
    }
    
    public String toString(){
        String status;
        if (speed!=OFF){
            status="Status:ON speed:"+speed+" location:"+location;
        }
        else{
            status="Status:Off"+ "location:"+location;
        }
        return status;
    }    
}

File: php/oop-study/lesson3/Undo/devices/CeilingFan.php
<?php
class CeilingFan{
	private $speed;//0 - выключен, 1 - медленно, 2 - средне, 3 - быстро
    private $location;
    public function __construct($location){
        $this->location=$location;
        $this->speed=0;
    }
    public function high(){
        $this->speed=3;
    }
    public function medium(){
        $this->speed=2;
    }
    public function low(){
        $this->speed=1;
    }
    public function off(){
        $this->speed=0;
    }
    public function getSpeed(){
        return $this->speed;
    } 
	public function status(){
		echo "------------------------------------------<br>";
		echo "Location:".$this->location."<br>";
		echo "Speed:".$this->speed."<br>";
		echo "------------------------------------------<br>";
	}
} 
?>